nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-administration/src/main/java/org/apache/nifi/admin/service/impl/StandardKeyService.java

```diff
@@ -119,10 +119,18 @@ public void deleteKey(String identity) {
 
             // delete the keys
             DeleteKeysAction deleteKeys = new DeleteKeysAction(identity);
-            transaction.execute(deleteKeys);
-
-            // commit the transaction
-            transaction.commit();
+            Integer rowsDeleted = transaction.execute(deleteKeys);
+
+            // commit the transaction if we found one and only one matching user identity
+            if(rowsDeleted == 0) {
+                rollback(transaction);
+                throw new AdministrationException("Unable to find user identity " + identity + " to remove token.");
+            } else if(rowsDeleted > 1) {
```

Comment:
Is there a legitimate scenario where there should be multiple identical identities in the database? If someone can take advantage of the identity mappings and cause their identity to match someone else's, this could be an attack vector.

Reply:
I don't believe so, however I thought it best to provide more feedback from the DAO. One problem with investigating this issue was that there was not enough information provided at this level, and was only really identified by accessing the actual queries to h2. By throwing exceptions for 0 or greater than 1, we can catch potential issues like this.